File: src/main/java/chat/teco/tecochat/common/presentation/PageResponse.java
package chat.teco.tecochat.common.presentation;

import java.util.List;
import org.springframework.data.domain.Page;

public record PageResponse<T>(
        List<T> content,
        boolean last,
        boolean first,
        boolean empty,
        int totalPages,
        int currentPages,
        long totalElements,
        int numberOfElements
) {

    public static <T> PageResponse<T> from(Page<T> response) {
        return new PageResponse<T>(
                response.getContent(),
                response.isLast(),
                response.isFirst(),
                response.isEmpty(),
                response.getTotalPages(),
                response.getNumber(),
                response.getTotalElements(),
                response.getNumberOfElements());
    }
}

File: src/main/kotlin/chat/teco/tecochat/support/ui/PageResponse.kt
package chat.teco.tecochat.support.ui

import org.springframework.data.domain.Page

data class PageResponse<T>(
    val content: List<T>,
    val last: Boolean,
    val first: Boolean,
    val empty: Boolean,
    val totalPages: Int,
    val currentPages: Int,
    val totalElements: Long,
    val numberOfElements: Int,
) {

    companion object {
        @JvmStatic
        fun <T> from(response: Page<T>): PageResponse<T> {
            return PageResponse(
                response.content,
                response.isLast,
                response.isFirst,
                response.isEmpty,
                response.totalPages,
                response.number,
                response.totalElements,
                response.numberOfElements
            )
        }
    }
}

